### ISSUE TITLE
Question about unsafe_encode_char()

### ISSUE THREAD

**Issue by ideal** [JuliaStrings/utf8proc#144]
```c
   } else if (uc == 0xFFFF) {
       dst[0] = (utf8proc_uint8_t)0xFF;
       return 1;
   } else if (uc == 0xFFFE) {
       dst[0] = (utf8proc_uint8_t)0xFE;
       return 1;
   }
```

Though U+FFFE and U+FFFF are both not characters (https://www.unicode.org/charts/PDF/UFFF0.pdf), why are they treated specially encoding to one byte in unsafe_encode_char()? Thanks.

**Comment by stevengj** [comment]
See #34.  These are ["noncharacters"](http://www.unicode.org/faq/private_use.html#nonchar1) and but we are required to encode them: *An implementation which converts noncharacter code points between one UTF representation and another must preserve these values correctly. The fact that they are called "noncharacters" and are not intended for open interchange does not mean that they are somehow illegal or invalid code points which make strings containing them invalid.*

However, the encoding here seems wrong to me — U+FFFF is supposed to encode to EFBFBF and U+FFFE is supposed to encode to EFBFBE, I thought.  I will look into it.

cc @ScottPJones, who was the last person to touch this code in #35.

**Comment by stevengj** [comment]
Actually, `utf8proc_encode_char` is correct for these codepoints.  It's only the internal `unsafe_encode_char` that is doing something odd, I forget why…

**Comment by stevengj** [comment]
Oh, that's right, we discussed this in #35.  The `CHARBOUND` feature internally uses 0xffff to insert grapheme markers as 0xff bytes.  But then this will get confused if there are actual 0xffff noncharacters in the string.

Nowadays we tend to recommend `utf8proc_grapheme_break_stateful` to find grapheme boundaries; we've talked about deprecating `CHARBOUND` for a while but haven't gotten around to it.
